### ISSUE TITLE
[BUG] - 322. Coin Change editorial

### ISSUE THREAD

**Issue by gyin** [LeetCode-Feedback/LeetCode-Feedback#26038]
### LeetCode Username

GerardYin

### Problem Number, Title, and Link

322. Coin Change https://leetcode.com/problems/coin-change

### Bug Category

Editorial

### Bug Description

Editorial Approach 3 explanation is confusing. Especially, the schema shows a matrix with little explanation on what those tables represent. I spent a lot of time thinking that we needed to materialize the left matrix, and it didn't make any sense. It was until I looked at the implementation that I understood that we were only materializing the right array, not the full matrix; and what the left matrix columns and row meant.

### Language Used for Code

None

### Expected behavior

I would suggest you make an annotated animation, or at the very least, you textually explain what is going on in this schema.

**Comment by gyin** [comment]
Hi there,

If you just add annotations to the diagram, make sure you make clear to the reader:
1. That only the right dp array is materialized (and not the left matrix), and this dp array in index i materializes the minimum amount of coins needed to reach i monetary value.
2. That every index of the dp array contains the number of minimum coins to reach that amount. And the fact it only contains integer indices is due to the fact that coins are integers. And that dp[i] once computed contains F(i)
3. What the left matrix is about. This matrix explains a lot of the confusion, because the column labels of the matrix are coins, and the content of every cell is not clear. The best I can describe it is the minimum number of coins to get dp[i] minus the value of the coin in the column label.
4. The arrow from the left matrix to the right matrix is not clear. I'd reword as "Take the minimum of the left values, and add one"

My explanations may not be the best ones, but I think I was not the only person confused (look at the editorial comment).

Instead I would suggest you make an animation, that would show layer by layer how dp[i] is made.

Gerard

**Comment by luciphania** [comment]
Thank you for your time.

Your feedback has been used to enhance the problem. As a token of our appreciation, your LeetCode account has been credited with 100 LeetCoins.

If you have any more questions or additional feedback, please don't hesitate to let us know. Your continued support is invaluable to us!

Best regards,
The LeetCode Team
